src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -4014,17 +4010,25 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                         if (embOp->Op(2)->IsCnsIntOrI())
                         {
                             MakeSrcContained(op2, embOp->Op(2));
+                            JITDUMP("Containing ShiftRight op2 inside ConditionalSelect\n");
+                            DISPTREERANGE(BlockRange(), op2);
                         }
                     }
                 }
 
                 // Handle op3
-                if (op3->IsVectorZero() && op1->IsMaskAllBitsSet())
+                if (op3->IsVectorZero() && op1->IsMaskAllBitsSet() && op2->IsEmbMaskOp())
                 {
                     // When we are merging with zero, we can specialize
                     // and avoid instantiating the vector constant.
                     // Do this only if op1 was AllTrueMask
                     MakeSrcContained(node, op3);
+                    if (op3->OperIsConvertMaskToVector())
```

Comment:
Wonder if we can go as far as putting the `OperIsConvertMaskToVector()` check inside `MakeSrcContained()` (and also one for `OperIsConvertVectorToMask()`)